### PR TITLE
set `enable_ecs_service_role` to `true` only if there is a load balancer

### PR DESCRIPTION
**Pull request by verbalius**
When creating service, and there is no LB defined "You cannot specify an IAM role for services that require a service linked role."
Set enable_ecs_service_role to false when there is 0 LB or more than 1.

## what
When creating service with network mode != awsvpc and there is no LB defined it throws error: "You cannot specify an IAM role for services that require a service linked.

## why
When setting value enable_ecs_service_role we check if there is no more than 1 LB, but less than 1 is 0 and we can't assign IAM role when it is no LB and network mode is bridge, host or none.

## solution
Set `enable_ecs_service_role` to `false` when there is 0 LB or more than 1 when network mode is not `awsvpc`

### CODE REVIEW

File: main.tf
Comment (nitrocode) [inline main.tf:3]: @verbalius the original code enables the `enable_ecs_service_role` if the `length(var.ecs_load_balancers) <= 1` meaning if there aren't any load balancers.

Your change sets enables the `enable_ecs_service_role` if the number of load balancers is equivalent to 1 which conflicts with the PR title. Is this intended ? Could you explain ?

Comment (verbalius) [reply main.tf:3]: Hi :)
I tried doing all the steps manually in the console and see
1. Task with bridge mode network
2.a. Service without LB (means length(var.ecs_load_balancers))
**Result**: You can't specify IAM role

image
2.b Service with application LB for example
**Result**: Lets you choose a role

image

So if we transfer it to expression:
if LB count = 0 this should be null:
`iam_role = local.enable_ecs_service_role ? coalesce(var.service_role_arn, join("", aws_iam_role.ecs_service.*.arn)) : null`
But how it is now `length(var.ecs_load_balancers)=0 <= 1 -> true`

Comment (verbalius) [reply main.tf:3]: or if you want to try it yourself via terraform just try to create service with bridge network mode without load balancer.

**Review by nitrocode — APPROVED** [review]

**Review by nitrocode — CHANGES_REQUESTED** [review]
We have to wait for the security group module to be complete before this can go through.

Sorry!

**Comment by nitrocode** [comment]
Blocked by https://github.com/cloudposse/terraform-aws-security-group/pull/17

File: main.tf
Comment (welderpb) [reply main.tf:3]: Why your commit contains: ```length(var.ecs_load_balancers) == 1``` ?

In our case we can define more than one load balancer for service..

**Comment by n0rig** [comment]
@nitrocode this PR (#118) should be closed and this one re-opened:
https://github.com/cloudposse/terraform-aws-ecs-alb-service-task/pull/137

The code change in this PR (#118) is wrong.

**Comment by verbalius** [comment]
Sorry for inactivity, added a fix commit as is in pull request #137. Thank you for help

**Comment by joe-niland** [comment]
@n0rig just FYI, we are still waiting on some related fixes to happen before merging this one

**Pull request closed without merging**
